src/assets/TestingWithTestCafe/test_cafe_blog.md

```diff
@@ -0,0 +1,162 @@
+
+## Brief about TestCafe
+Let's start this blog by understanding a brief about `test-cafe`. It is an open-source test automation framework or tool built with Node Js. This framework basically supports 2 programming languages i.e javascript and Typescript. So you are required to have at least basics of Javascript to use this tool. [DevExpress](https://www.devexpress.com/) is in charge of managing TestCafe, which is made available under an open-source MIT license.
+
+## Why to use TestCafe ?
+Before moving to installation and set-up, lets understand some points why to use `test-cafe`.
+
+- It is very easy to set up
+- No dependencies (with other libraries)
+- writing test is easy and with less code
+- Cross-browser testing is made easy
+- Free and Open Source
+
+*Note: I will doing the whole setup with Ubuntu 20.04.3 LTS, but it will be similar for other OS too*
+
+## Prerequisites
+
+At first, we need `Node JS` installed into our system. You can download `Node JS` version 12 or above. [Download Node JS](https://nodejs.org/en/download/).
+
+## Installation and setup
+Now lets move into installing `test-cafe` which is really easy. Follow the following steps:
+
+- Create a folder `E2EWithTestCafe`.
+- Open the folder you created with a code editor. In my case I am using `Visual Studio Code`.
+- Create a `package.json` file with command `npm init` through command line. (using terminal)
+- Now use `npm install testcafe` command to install TestCafe.
+
+After this, your `package.json` file should contain `dependencies` as
+
+```json
+// package.json file
+
+"dependencies": {
+"testcafe": "^2.1.0"
+}
+```
+
+## What to test ?
+
+For this blog we will be writing E2E test for this `https://devexpress.github.io/testcafe/example` website (provided by devexpress). You can go to this website and check what does it look like. We will automate a simple form submit scenarios using `test-cafe`
+
+##  Writing Test
+
+At first inside `E2EWithTestCafe` project, make your file structure as
+
+```
+📦tests
+┗ 📂e2e
+  ┣ 📂fixtures
+    ┗ 📜firste2etest.js
+
+```
+And then at the root level of your project which is `E2EWithTestCafe` create a file `.testcaferc.json`. This file includes all the configuration required to test the application.
+
+The configuration should look like this:
+```json
+// testcaferc.json
+
+{
+  "browsers": "chrome", // means your test will be automated in chrome browser
+  "baseUrl": "https://devexpress.github.io/testcafe/example/", // URL for your web application to be tested or automated
+  "src": "tests/e2e/fixtures/firste2etest.js" // path to your test code to execute
+}
+
+```
+
+I assume that you have checked `https://devexpress.github.io/testcafe/example` website. In that website if we fill up the username and submit the form, it redirects us to another page saying thank you. So this will be our simple testcase, and we will be automating it with `test-cafe`
+I have implemented the test code And it looks like this.
```

Comment:
```suggestion
I assume that you have checked `https://devexpress.github.io/testcafe/example` website. On that website, if we fill up the username and submit the form, it redirects us to another page saying thank you. So this will be our simple test case, and we will be automating it with `test-cafe`
I have implemented the test code And it looks like this.
```